src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -80,11 +85,19 @@ public abstract class JsonPropertyInfo
         public Action<object, object?>? Set
         {
             get => _untypedSet;
-            set => SetSetter(value);
+            set
+            {
+                VerifyMutable();
+                SetSetter(value);
+                // Invalidate any JsonIgnore configuration if delegate set manually by user
+                IgnoreNullTokensOnRead = false;
+                _isUserSpecifiedSetter = true;
```

Comment:
will `property.Set = property.Set` cause null tokens to not be ignored?

Reply:
consider storing original reference rather than bool flag

Reply:
As a matter of fact, this invalidation is no longer needed -- since I reverted the behavior of `IgnoreNullTokens` it is only possible for this flag to be set during the `Configure` stage. I'll remove that.